test/test_cases.cpp

```diff
@@ -16,6 +16,53 @@
 
 namespace ekustd {
 
+class IChar {
```

Comment:
@glpuga this one's clever! I wonder if by template-ing its base class you could foster reuse.